File: framework/scala/src/main/scala/soda/scala/unittest/task/TaskProxy.scala
package soda.scala.unittest.task

import soda.scala.unittest.WorkInput

import scala.reflect.runtime.universe.Type

trait TaskProxy[R] {

  def returnType: Type

  def argumentTypes: List[Type]

  def arguments: List[Any]

  def execute(input: WorkInput): R

  def elapseMillis: Double

}

File: framework/java/src/main/java/soda/unittest/task/TaskProxy.java
package soda.unittest.task;

import soda.unittest.WorkInput;

import java.lang.reflect.Type;

public interface TaskProxy<R> {

    Type getReturnType();

    Type[] getArgumentTypes();

    Object[] getArguments();

    R execute(WorkInput input);

    double getElapseMillis();

}
